### PR TITLE
fix create dedciated elb with eip options issue

### LINKED ISSUE #208: Issue when creating a dedicated ELB with auto create EIP
**What happened**:
Trying to create a ELB on a partner cloud by using Huawei Cloud K8s CCM, i choose dedicated elb and auto create EIP
x
**What you expected to happen**:
ELB should be created with EIP attached

**How to reproduce it (as minimally and precisely as possible)**:
Create ELB using CCM and provide following 
metadata:
  annotations:
    kubernetes.io/elb.class: dedicated
    kubernetes.io/elb.availability-zones: ae-ad-1a
    kubernetes.io/elb.l4-flavor-id: a53df111-cc06-4c49-a417-2aac536f8eb3
    kubernetes.io/elb.lb-algorithm: ROUND_ROBIN
    kubernetes.io/elb.keep-eip: "false"
    kubernetes.io/elb.eip-auto-create-option: >-
      {"ip_type": "5_bgp", "bandwidth_size": 5, "share_type": "PER"}

**Anything else we need to know?**:
If i don't provide this option, ELB gets created without EIP attached. I have also modified the 
      {"bandwidth_name":"some_name","ip_type": "5_bgp", "bandwidth_size": 5, "share_type": "PER"}
 and 
      {"name":"some_name","ip_type": "5_bgp", "bandwidth_size": 5, "share_type": "PER"}

**Environment**:
- Kubernetes version (use `kubectl version`): 1.24.11
- Cloud provider or hardware configuration: Huawei Cloud 
- OS (e.g: `cat /etc/os-release`): ubuntu20.04 LTS
- Kernel (e.g. `uname -a`): 5.4.0-99-generic
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

Logs from CCM:
E0313 12:37:42.110722       1 dedicated_elb.go:47] Error in wrapper handler(), args: []interface {}{"Loadbalancer", (**model.LoadBalancer)(0xc000514c60)}, error: {"status_code":400,"request_id":"1402655ec2c66d91dd28901d69ec3d20","error_code":"ELB.8902","error_message":"Unable to find 'name' required attribute in bandwidth when id is null."}
I0313 12:37:42.110748       1 controller.go:808] Finished syncing service "take11/loadbalancer-service-demo" (4.515507678s)
E0313 12:37:42.110762       1 controller.go:289] error processing service take11/loadbalancer-service-demo (will retry): failed to ensure load balancer: {"status_code":400,"request_id":"1402655ec2c66d91dd28901d69ec3d20","error_code":"ELB.8902","error_message":"Unable to find 'name' required attribute in bandwidth when id is null."}
I0313 12:37:42.110803       1 event.go:294] "Event occurred" object="take11/loadbalancer-service-demo" fieldPath="" kind="Service" apiVersion="v1" type="Warning" reason="SyncLoadBalancerFailed" message=<
        Error syncing load balancer: failed to ensure load balancer: {"status_code":400,"request_id":"1402655ec2c66d91dd28901d69ec3d20","error_code":"ELB.8902","error_message":"Unable to find 'name' required attribute in bandwidth when id is null."}

### PR DESCRIPTION
**Pull request by cloudomate**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
To fix create dedicated elb with eip options, bandwidth name is required when share_type is PER

**Which issue(s) this PR fixes**:

Fixes #208 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Yes, additional field is required to be filled by the user for configuring elb with create EIP with options, the user guide has also been updated as a part of PR
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by chengxiangdong** [comment]
@cloudomate 
Thanks for your contribution, there are some lint issues to fix before merging.

File: docs/huawei-cloud-controller-manager-configuration.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ This section provides Huawei Cloud IAM configuration and authentication informat
 
 * `secret-key` Required. The secret key of the Huawei Cloud.
 
-* `project-id` Optional. The Project ID of the Huawei Cloud. 
+* `project-id` Optional. The Project ID of the Huawei Cloud. If cloud field is provided and is not `myhuaweicloud.com`, this field is mandatory
```
Comment (chengxiangdong) [inline docs/huawei-cloud-controller-manager-configuration.md:44]: There is an error in the published document, this field is required.

File: docs/huawei-cloud-controller-manager-configuration.md [hunk 2]
```diff
@@ -54,9 +54,9 @@ This section provides Huawei Cloud IAM configuration and authentication informat
 
 This section contains network configuration information.
 
-* `id` Optional. Specifies the VPC used by ECSes of the Kubernetes cluster.
+* `id` Optional. Specifies the VPC used by ECSes of the Kubernetes cluster. If cloud field under global section is provided and is not `myhuaweicloud.com`, this field is mandatory
```
Comment (chengxiangdong) [inline docs/huawei-cloud-controller-manager-configuration.md:57]: This field is not used currently

File: docs/huawei-cloud-controller-manager-configuration.md [hunk 3]
```diff
@@ -54,9 +54,9 @@ This section provides Huawei Cloud IAM configuration and authentication informat
...
 
-* `subnet-id` Optional. Specifies the IPv4 subnet ID used by ECSes of the Kubernetes cluster.
+* `subnet-id` Optional. Specifies the IPv4 subnet ID used by ECSes of the Kubernetes cluster. If cloud field under global section is provided and is not `myhuaweicloud.com`, this field is mandatory
```
Comment (chengxiangdong) [inline docs/huawei-cloud-controller-manager-configuration.md:59]: Shared ELB is optional, while dedicated ELB is required.

File: docs/usage-guide.md [hunk 4]
```diff
@@ -49,6 +49,8 @@ otherwise use the set value.
 
   * `bandwidth_size` Required. Specifies the bandwidth size.
 
+  * `bandwidth_name` Required when share_type is PER. Specifies the bandwidth name and is mandatory for Dedicated Loadbalancer if share_type is per.
```
Comment (chengxiangdong) [inline docs/usage-guide.md:52]:
```suggestion
  * `bandwidth_name` Optional, it is required when `share_type=PER`. Specifies the bandwidth name.
```

**Review by chengxiangdong — COMMENTED** [review]
1. The bandwidth name parameter becomes mandatory due to API changes.
2. `project-id` should be mandatory.
3. `subnet-id` is optional.
4. VPC ID is not used currently.

File: docs/usage-guide.md (hunk 4 above)
Comment (cloudomate) [reply docs/usage-guide.md]: done in two commits as i read your message later

**Comment by chengxiangdong** [comment]
https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/blob/82721352c3cddf23f1941e33c9bc38b2ad734987/pkg/cloudprovider/huaweicloud/sharedloadbalancer.go#L1011

When creating bandwidth for a shared ELB, the bandwidth name is the service name .
Can you change this to also use `bandwitd_name`?

**Comment by cloudomate** [comment]
> https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/blob/82721352c3cddf23f1941e33c9bc38b2ad734987/pkg/cloudprovider/huaweicloud/sharedloadbalancer.go#L1011
> 
> When creating bandwidth for a shared ELB, the bandwidth name is the service name . Can you change this to also use `bandwitd_name`?
you mean pick up the bandwidth name from the ELB name.

**Comment by chengxiangdong** [comment]
> https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/blob/82721352c3cddf23f1941e33c9bc38b2ad734987/pkg/cloudprovider/huaweicloud/sharedloadbalancer.go#L1011
> 
> When creating bandwidth for a shared ELB, the bandwidth name is the service name . Can you change this to also use `bandwitd_name`?
> you mean pick up the bandwidth name from the ELB name.

It's a good idea to add the bandwidth name to parameters.

Before this, the shared ELB used the service name as the bandwidth name, and the dedicated ELB caused bugs because the `bandwidth_name` was not set.
Now that the bug has been fixed, and the dedicated ELB bandwidth name also uses the service name.

For compatibility, if andwidth_name annotation is not specified , the service name should continue to be used as the bandwidth name when creating bandwidth or EIP.

**Pull request closed without merging**